frontend/test/account-send.test.ts

```diff
@@ -0,0 +1,561 @@
+    // of the values.
+    await localforage.setItem(localStorageValueKey, [
+      {
+        accountSendCiphertext: '',
+        amount: '100',
+        tokenAddress,
+        dateSent: new Date(),
+        txHash,
+      },
+    ]);
+    const count = await localforage.getItem(localStorageCountKey);
+    expect(count).toEqual(undefined);
+
+    await storeSend(storeSendArgs);
+    const newCount = await localforage.getItem(localStorageCountKey);
+    const values = (await localforage.getItem(localStorageValueKey)) as any[];
+
+    expect(newCount).toBeTruthy();
+    expect(values.length).toEqual(1);
+  });
+
+  it('Preserves existing value and count when storing a new entry', async () => {
+    // These values are not representative
+    // of actual values but this test should work regardless
+    // of the values.
+    await localforage.setItem(localStorageValueKey, [
+      {
+        accountSendCiphertext: '',
+        amount: '100',
+        tokenAddress,
+        dateSent: new Date(),
+        txHash,
+      },
+    ]);
+    await localforage.setItem(localStorageCountKey, 10012);
+
+    await storeSend(storeSendArgs);
+
+    const count = await localforage.getItem(localStorageCountKey);
+    const values = (await localforage.getItem(localStorageValueKey)) as any[];
+    expect(count).toEqual(10012);
+    expect(values.length).toEqual(2);
+  });
+});
+
+describe('fetchAccountSends', () => {
+  const testingUtils = generateTestingUtils({ providerType: 'default' });
+
+  beforeEach(async () => {
+    await localforage.clear();
+    testingUtils.clearAllMocks();
+    window.logger = {
+      debug: jest.fn(),
+      info: jest.fn(),
+      warn: jest.fn(),
+      version: '',
+      _log: jest.fn(),
+      makeError: jest.fn(),
+      assert: jest.fn(),
+      assertArgument: jest.fn(),
+      checkNormalize: jest.fn(),
+      checkArgumentCount: jest.fn(),
+      checkNew: jest.fn(),
+      checkAbstract: jest.fn(),
+      checkSafeUint53: jest.fn(),
+      throwError: jest.fn() as never,
+      throwArgumentError: jest.fn() as never,
+    };
+  });
+
+  it('Correctly fetch send data when there is a single send', async () => {
+    await localforage.setItem(localStorageValueKey, [
+      {
+        accountSendCiphertext: '0xb609426a8909759990b22756b7f2ce4f8d5ac4685d6e2c40daa830d059950504',
+        amount: '100',
+        tokenAddress,
+        dateSent: new Date(),
+        txHash,
+      },
+    ]);
+    await localforage.setItem(localStorageCountKey, 2);
+
+    const accountSends = await fetchAccountSends({
+      address: recipientAddress,
+      viewingPrivateKey,
+      chainId: 5,
+    });
+
+    expect(accountSends.length).toEqual(1);
+    expect(accountSends[0].recipientAddress).toEqual(recipientAddress);
+    expect(accountSends[0].advancedMode).toEqual(true);
+    expect(accountSends[0].usePublicKeyChecked).toEqual(false);
+  });
+
+  it('Correctly fetch send data when there are multiple send', async () => {
```

Comment:
Can we add an integration test that starts from a blank local storage state and uses random values to store/read up to `n` entries. Something like this:
1. Generate a random `n` between 2 and 100
2. Generate random send data to save
3. Save it
4. Read it and ensure it matches the data from 2
5. Repeat steps 2-4 `n` times
6. Randomly clear localstorage periodically, e.g. maybe each iteration has a 25% chance of clearing it

We've had a few bugs pop up in this PR so we should be cautious here and make sure we have good/strong tests

Reply:
And then maybe the test repeats the above loop `NUM_RUNS` times

Reply:
https://github.com/ScopeLift/umbra-protocol/pull/494/commits/66ebbc25cdb80e202267af3fe4e119ff7f3f4b11 I implemented this slightly differently than the above.

I had to limit `n` to 20 because it was causing timeouts locally and I didn't see any obvious optimizations. I tried making the test more concurrent, but that caused some issues because of the way we handle encryption count.

And I read `NUM_RUNS` as re-running the test multiple times with a new `n` rather than maintaining the state and re-running with a new n.